### ISSUE TITLE
Add support for Dockerfile

### ISSUE THREAD

**Issue by fspacek** [eclipse/microprofile-starter#96]
Hi, 

It would be great if there is option to generate Dockerfile for selected application server based on official images. 

Basically it would copy war to deployment directory and copy generated server configuration.

**Comment by rdebusscher** [comment]
Hi František,

Using the official Docker Images will not (always) be possible. The Payara Docker Image, for example, allows you to run a WAR file, but the MicroProfile Starter is built around creating JAR files.

We can create a DockerFile which starts from a Java-based image but is this useful?

**Comment by fspacek** [comment]
Hello Rudy,

good point, however I think it can be useful even in case of pure Java based image.  

Such Dockerfile can be showcase of "best practices" how to bundle microprofile apps with proper JVM flags (important for Java 8). 

Just curious support of WARs is not planned?

**Comment by rdebusscher** [comment]
Most people find that MicroServices needs to be run from executable JAR files. They find that WARs are to tight linked to the 'old school' Application servers. (opinion of some, maybe not the truth :) )
However, when using WAR files, the docker Image Layer structure can be optimized.

MicroProfile Starter generates for most implementations a WAR file and afterward, it is packaged within an executable JAR.

**Comment by rdebusscher** [comment]
Can you give a list of JVM parameters which you want to see? Because there exist also a lot of different opinions about that. 

Then we can get the discussion started.

**Comment by Emily-Jiang** [comment]
I was thinking along the similar line. I think it is a good idea to provide a docker image for each runtime since we are building cloud-native microservices.

**Comment by fspacek** [comment]
Yes that was my point, that with WARs docker images can be pretty efficient. 

Regarding JVM flags for java 8 (from update 131) at least two are needed 
`-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap`.

From java 10+ it should be set correctly by default without any flags. 

There is also possibility to fine grain heap allocation with `-XX:(Min|Max|Initial)RAMPercentage`. 

[Reference](https://medium.com/adorsys/jvm-memory-settings-in-a-container-environment-64b0840e1d9e)

**Comment by asept** [comment]
Hi, 

i think, it‘s a good idea to have also a kubernetes configuration optional. 

For my company, it’s very important to have small war files (kB) to shipping with docker...
